File: src/main/java/com/timepath/hl2/io/util/Vector3f.java
package com.timepath.hl2.io.util;

import com.timepath.io.struct.StructField;

/**
 * @author TimePath
 */
public class Vector3f {

    @StructField(index = 0)
    private float x;
    @StructField(index = 1)
    private float y;
    @StructField(index = 2)
    private float z;

    public Vector3f() {
    }

    public Vector3f(float x, float y, float z) {
        this.x = x;
        this.y = y;
        this.z = z;
    }

    @Override
    public String toString() {
        return String.format("vec3(%s, %s, %s)", x, y, z);
    }
}

File: src/main/kotlin/com/timepath/hl2/io/util/Vector3f.kt
package com.timepath.hl2.io.util

import com.timepath.io.struct.StructField

public class Vector3f(x: Float = 0f, y: Float = 0f, z: Float = 0f) {
    StructField(index = 0)
    private val x: Float = x
    StructField(index = 1)
    private val y: Float = y
    StructField(index = 2)
    private val z: Float = z

    override fun toString() = "vec3($x, $y, $z)"
}
